### DISCUSSION TITLE
Does DocTR run on GPU or CPU?

### DISCUSSION THREAD

**Discussion by Shiva-Ai4m** [mindee/doctr#1462] [General]
@felixdittrich92 @felixT2K I wwanted to ask whetherthe DocTR Model runs on CPU or GPU? and Which gpu has best computation for docTR as i wanted to implement it for a condition for frame rate of 1 fps

**Comment by felixdittrich92** [comment]
Hi @Shivashankarar :wave:

which backend do you use PyTorch or TensorFlow ? (You can run on CPU or GPU (if you have an nvidia gpu))

1 fps is this correct ? ^^ So if you depend on inference latency the fastest way would be to use `db_mobilenet_v3_large` as detection model on GPU with half precision 

Example:

```
model = ocr_predictor(
    det_arch="db_mobilenet_v3_large",
    reco_arch="crnn_vgg16_bn",
    pretrained=True,
    det_bs=8,  # depends on your available VRAM
    reco_bs=1024,  # depends on your available VRAM
    assume_straight_pages=True,
    straighten_pages=False,
    detect_orientation=False,
).cuda().half()  # run on GPU with full float16 (half precision)
```

With a RTX 4080 this takes ~ 1.5sec for 14 images
